# Analysis of fremont bridge data

Ziah D.

10/17/19

## Abstract


For this lab I am taking the number of people that crossed the West sidewalk and the number of people that crossed the East sidewalk and then finding the Mean, median, and average of the data. Afterwards I am Reflecting on my discoveries and asking questions that were raised.

## Dataset Preparation

I got this file from Ms. Sconyers. This dataset has a total of 31 rows and 3 columns that orignally came from a dataset, measuring hourly bicycle data. Then the data was processed to produce daily bike data. The original data was from https://data.seattle.gov/Transportation/Fremont-Bridge-Hourly-Bicycle-Counts-by-Month-Octo/65db-xm6k.

I am writing the code below becuase I want to anlayze the data given about the amount of people that crossed the the Fremont Bridge West Sidewalk and the Fremont Bridge East Sidewalk. The first line of code allows me to open the code on jupyter notebook. One reason why I added the encoding part in the code is to make sure I read the filing correctly. The second line  creates an empty list to store data from the file that we opened on the Fremont bridge data .The third line of code is putting all lines from the file into a list. The fourth line of data is closing the Bridge file, closing the file is a corteous and clean thing to do. The last line is a representation of what the first 5 rows of code look like, giving people an idea of what it looks like.


The code block below opens a cvs file with data that we then copy into a list. Lastly, we print some rows of data so that people reading the code can get a sense of waht it looks like and so that we can make sure that the list is working.

In [31]:
#list of number of people that walked across the west and east sidewalk

#open the file
Fremont_file = open("Fremont_bridge_data (1).csv", "r", encoding = "ISO-8859-1");

#create an empty list to store the data from the csv file
Fremont_data = []; 

#put all lines from the file into a list
for row in Fremont_file:
    Fremont_data.append(row);
    
#close the file. It's the courteous and clean thing to do
Fremont_file.close();

#this is what five rows of the data look like-helps give people an idea of what it looks like
print(Fremont_data[:5]);



['Date,Fremont Bridge West Sidewalk,Fremont Bridge East Sidewalk\n', '9/1/15,1393,1483\n', '9/2/15,1745,1867\n', '9/3/15,1532,1627\n', '9/4/15,1425,1512\n']


The list above represents the first five rows in the file communities_data. I printed the first five rows in order to let people know what the data looks like.

In the section below I am creating two empty lists, and then deleting the first row , which consists of headers. Next, I take the data and split it based off of where the commas are. Lastly, I count like a data scinetist, 0,1,2 in order to pick the right collumn of data. I'm taking all of the of the information in the 2nd collumn and turning it into an integer, then adding it to the Number_of_West_People list. And, i'm taking all of the information from  the 3rd collumn and turning it into an integer, and then adding it to the Number_of_East_People list.

In [32]:
#remove the first element (column header)
# the header is useful for telling us what the columns mean, 
# but you don't want to include it in your mathematical calculations
del Fremont_data[0];

# create two empty lists for number of people that go across on the east and westside of the fremont bridge.
Number_of_East_People = []; 
Number_of_West_People = [];


# loop (go through every row one by one) through the original fremont bridge data
for item in range(len(Fremont_data)):
    
    # this line is a bit confusing, but what it is doing is splitting out each comma separated item 
    # from a row and turning it into a list 
    # for example, this line of code turns he single string '9/1/15,1393,1483\n' 
    # into three separate strings '9/1/15' '1393' and '1483'
    # After being split, the three separate strings are then stored in another list called Fremont_row
    Fremont_row = Fremont_data[item].split(",")
    
    #now that we've split the row into its three parts, we can pick and choose 
    Number_of_West_People.append(int(Fremont_row[1]))
    
    Number_of_East_People.append(int(Fremont_row[2]))
    

## Data Modeling

Now that we have these lists I am able to do calculations on the minimum,maximum, average, and median on the number of bikes crossing on a daily basis for October, 2012. I am doing these calculations because I want to find those numbers so that I can reflect on the results.

In [33]:
#create variables for the minimum, maximum, average, and median.

#max, min, average, median ,and desriptions

#Using import statistics which has a code that finds the mean and median

import statistics

#creating varibales for certain for mathmatical calculations including finding the Minimum, maximim, mean, and median for east data
Eastnumbmin = min(Number_of_East_People)

Eastnumbmax = max(Number_of_East_People)

Eastnumbavg = statistics.mean(Number_of_East_People)

Eastnumbmed = statistics.median(Number_of_East_People)


#creating varibales for certain for mathmatical calculations including finding the Minimum, maximim, mean, and median for west data.
Westnumbmin = min(Number_of_West_People)

Westnumbmax = max(Number_of_West_People)

Westnumbavg = statistics.mean(Number_of_West_People)

Westnumbmed = statistics.median(Number_of_West_People)





#print all the data for both east and west, and their titles

print("minimum number of people going east:")
print(Eastnumbmin)
print("maximum number of people going east:")
print(Eastnumbmax)
print("average number of people going east:")
print(Eastnumbavg)
print("median number of all people going east:")
print(Eastnumbmed)


print("minimum number of people going west:")
print(Westnumbmin)
print("maximum number of people going west:")
print(Westnumbmax)
print("average number of people going west:")
print(Westnumbavg)
print("median number of all people going:")
print(Westnumbmed)


minimum number of people going east:
425
maximum number of people going east:
2361
average number of people going east:
1540.4333333333334
median number of all people going east:
1648.0
minimum number of people going west:
466
maximum number of people going west:
2270
average number of people going west:
1497.5666666666666
median number of all people going:
1573.5


### conclusions:
The minimum number that I got for the dataset on the amount of bikers that went across the fremont bridge is 425 people, going east, and 466 going west. Since I have a fairly small dataset, I am able to look back on the exact date both of those numbers came from. It looks like both of those numbers came from the september 6th. I looked up online what day of this week this day was and it said it was a sunday. This raises numerous questions which will be shown below. Additionally I can conclude that on this particular day there was less bike traffic on the fremont bridge. Additionally, I found that the maxmum number of bikers crossing the bridge,unlike the minimum were on seperate days. The maximum number of bikers going East was the day before the maximum number of bikers going west. The last numbers that I found was important was the median and average number of bikers going west and east. I found that the average number of bikers going west is around 1497 and the average number going east is around 1540. Knowing this allows me to conclude that on average there were more bikers going east. Lastly, I found the median numbers for the bikers were less important, but the numbers were, 1648 people going going east and around 1573 going west.

### questions raised:

The minimum number of bikes that crossed on either side was on a Sunday, what is special about sunday?

Was the bridge shut down at one point for that day/ or was the number just really small that day?

Was there any construction throughout the collection data?




## Acknowledgements

For the second lab on the analysis of the fremont bike data I got assistance from Ms. Sconyers and Navya.